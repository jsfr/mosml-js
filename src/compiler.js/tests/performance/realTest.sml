prim_val ctimebegin_ : string -> unit = 1 "console.time";
prim_val ctimeend_ : string -> unit = 1 "console.timeEnd";
fun ctimeBegin n = ctimebegin_ n;
fun ctimeEnd n = ctimeend_ n;

fun repeat n f x =
  let 
    fun loop 0 = f x
      | loop n = (f x; loop(n-1))
  in 
    loop(n-1) 
  end 
fun timerep n f s = repeat n (fn x => (ctimeBegin(s);f x; ctimeEnd(s)))


fun realAdd x =
  let
    val addRef = ref 0.0;
  in
    while !addRef < x do (addRef := !addRef+1.0; 33.3334+9.992)
  end

fun realSub x =
  let
    val subRef = ref 0.0;
  in
    while !subRef < x do (subRef := !subRef+1.0; 33.3334-9.992)
  end
  

val it = timerep 100 realAdd "realAdd" 10000000.0
val it = timerep 100 realSub "realSub" 10000000.0




