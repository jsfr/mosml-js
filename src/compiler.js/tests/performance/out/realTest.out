beautified realTest.js
==============================JS BEGIN==============================
addInt: 80ms
addInt: 78ms
addInt: 78ms
addInt: 79ms
addInt: 95ms
addInt: 148ms
addInt: 106ms
addInt: 82ms
addInt: 79ms
addInt: 77ms
addInt: 78ms
addInt: 85ms
addInt: 88ms
addInt: 85ms
addInt: 84ms
addInt: 85ms
addInt: 85ms
addInt: 84ms
addInt: 85ms
addInt: 85ms
addInt: 84ms
addInt: 85ms
addInt: 84ms
addInt: 88ms
addInt: 84ms
addInt: 85ms
addInt: 84ms
addInt: 89ms
addInt: 90ms
addInt: 93ms
addInt: 86ms
addInt: 83ms
addInt: 87ms
addInt: 85ms
addInt: 84ms
addInt: 86ms
addInt: 80ms
addInt: 79ms
addInt: 79ms
addInt: 79ms
addInt: 79ms
addInt: 79ms
addInt: 79ms
addInt: 81ms
addInt: 79ms
addInt: 78ms
addInt: 79ms
addInt: 88ms
addInt: 85ms
addInt: 88ms
addInt: 86ms
addInt: 89ms
addInt: 89ms
addInt: 90ms
addInt: 86ms
addInt: 87ms
addInt: 85ms
addInt: 88ms
addInt: 85ms
addInt: 86ms
addInt: 85ms
addInt: 88ms
addInt: 87ms
addInt: 87ms
addInt: 85ms
addInt: 87ms
addInt: 86ms
addInt: 87ms
addInt: 85ms
addInt: 84ms
addInt: 83ms
addInt: 80ms
addInt: 79ms
addInt: 79ms
addInt: 81ms
addInt: 85ms
addInt: 83ms
addInt: 87ms
addInt: 79ms
addInt: 79ms
addInt: 79ms
addInt: 86ms
addInt: 85ms
addInt: 86ms
addInt: 86ms
addInt: 85ms
addInt: 87ms
addInt: 85ms
addInt: 86ms
addInt: 85ms
addInt: 86ms
addInt: 86ms
addInt: 85ms
addInt: 87ms
addInt: 86ms
addInt: 85ms
addInt: 87ms
addInt: 86ms
addInt: 89ms
addInt: 90ms
subInt: 95ms
subInt: 85ms
subInt: 84ms
subInt: 85ms
subInt: 85ms
subInt: 85ms
subInt: 85ms
subInt: 84ms
subInt: 85ms
subInt: 85ms
subInt: 84ms
subInt: 85ms
subInt: 87ms
subInt: 84ms
subInt: 88ms
subInt: 83ms
subInt: 84ms
subInt: 84ms
subInt: 84ms
subInt: 84ms
subInt: 85ms
subInt: 86ms
subInt: 91ms
subInt: 91ms
subInt: 88ms
subInt: 84ms
subInt: 85ms
subInt: 85ms
subInt: 80ms
subInt: 79ms
subInt: 79ms
subInt: 78ms
subInt: 78ms
subInt: 79ms
subInt: 79ms
subInt: 80ms
subInt: 79ms
subInt: 78ms
subInt: 81ms
subInt: 87ms
subInt: 84ms
subInt: 84ms
subInt: 84ms
subInt: 85ms
subInt: 88ms
subInt: 102ms
subInt: 159ms
subInt: 109ms
subInt: 85ms
subInt: 85ms
subInt: 86ms
subInt: 85ms
subInt: 85ms
subInt: 85ms
subInt: 84ms
subInt: 85ms
subInt: 85ms
subInt: 84ms
subInt: 86ms
subInt: 85ms
subInt: 85ms
subInt: 84ms
subInt: 86ms
subInt: 84ms
subInt: 86ms
subInt: 85ms
subInt: 85ms
subInt: 102ms
subInt: 152ms
subInt: 89ms
subInt: 89ms
subInt: 84ms
subInt: 84ms
subInt: 84ms
subInt: 85ms
subInt: 85ms
subInt: 84ms
subInt: 85ms
subInt: 86ms
subInt: 85ms
subInt: 84ms
subInt: 86ms
subInt: 85ms
subInt: 88ms
subInt: 85ms
subInt: 87ms
subInt: 85ms
subInt: 85ms
subInt: 86ms
subInt: 86ms
subInt: 138ms
subInt: 99ms
subInt: 92ms
subInt: 85ms
subInt: 85ms
subInt: 85ms
subInt: 87ms
subInt: 85ms
subInt: 86ms
subInt: 85ms
==============================JS END================================
